src/Transformations.jl

```diff
@@ -107,4 +107,50 @@ function find_diag_axes(x::AbstractArray, atol=1e-12)
     end
 end
 
+Base.@kwdef struct AntiDiagonalGauging <: Transformation
+    atol::Float64 = 1e-12
+    skip::Vector{Index} = Symbol[]
+end
+
+function transform!(tn::TensorNetwork, config::AntiDiagonalGauging)
+    skip_inds = isempty(config.skip) ? openinds(tn) : config.skip
+    queue = collect(keys(tn.tensors))
+
+    while !isempty(queue) # loop over all tensors
+        idx = pop!(queue)
```

Comment:
Why don't you just do a `for` loop over the `keys(tn.tensors)`?

```suggestion

    for idx in keys(tn.tensors)
```